models/baseModels/Invoice/Invoice.ts

```diff
@@ -166,6 +177,52 @@ export abstract class Invoice extends Transactional {
     await partyDoc.updateOutstandingAmount();
   }
 
+  async _updateReturnInvoiceOutStanding() {
+    if (!this.isReturn || !this.returnAgainst || !this.grandTotal) {
+      return;
+    }
+
+    const invoiceOutstandingAmount = (
+      Object.values(
+        await this.fyo.db.get(
+          this.schema.name,
+          this.returnAgainst as string,
+          'outstandingAmount'
+        )
+      )[0] as Money
+    ).abs();
+
+    const returnInvoiceGrandTotal = this.grandTotal.abs();
+    let outstandingAmount;
+
+    if (this.isSubmitted) {
+      if (invoiceOutstandingAmount.isZero()) {
+        return;
+      }
+      outstandingAmount = invoiceOutstandingAmount.sub(returnInvoiceGrandTotal);
+    }
+
+    if (this.isCancelled) {
+      outstandingAmount = invoiceOutstandingAmount.add(returnInvoiceGrandTotal);
+    }
+
+    this.fyo.db.update(this.schema.name, {
+      name: this.returnAgainst,
+      outstandingAmount,
+    });
+  }
+
+  async _updateReturnCompleteStatus() {
+    if (!this.isReturn && !this.returnAgainst) {
+      return;
+    }
+
+    this.fyo.db.update(this.schemaName, {
```

Comment:
Don't use `db.update`, set and sync. `db.update` will throw the doc out of sync.